src/replica/replica_stub.cpp

```diff
@@ -1568,175 +1483,6 @@ void replica_stub::on_gc_replica(replica_stub_ptr this_, gpid id)
-}
-
-void replica_stub::flush_replicas_for_slog_gc(const replica_gc_info_map &replica_gc_map,
-                                              const std::set<gpid> &prevent_gc_replicas)
-{
-    // Trigger checkpoints to flush memtables once some replicas were found that prevent slog files
-    // from being removed for gc.
-    //
-    // How to trigger memtable flush ?
-    //   A parameter `is_emergency' was added for `replica::background_async_checkpoint()` function;
-    //   once it's set true, underlying storage engine would flush memtable as soon as possiable.
-    //
-    // When memtable flush is triggered ?
-    //   1. After a fixed interval (specified by `[replication].gc_interval_ms` option), try to find
-    //   if there are some replicas preventing slog files from being removed for gc; if any, all of
-    //   them would be deleted "gradually" ("gradually" means the number of the replicas whose
-    //   memtables are submitted to storage engine to be flushed would be limited).
-    //   2. `[replication].checkpoint_max_interval_hours' option specified the max interval between
-    //   the two adjacent checkpoints.
-
-    if (prevent_gc_replicas.empty()) {
-        return;
-    }
-
-    limit_flush_replicas_for_slog_gc(prevent_gc_replicas.size());
-    _last_prevent_gc_replica_count = prevent_gc_replicas.size();
-
-    LOG_INFO("gc_shared: trigger emergency checkpoints to flush replicas for gc shared logs: "
-             "log_shared_gc_flush_replicas_limit = {}/{}, prevent_gc_replicas({}) = {}",
-             _real_log_shared_gc_flush_replicas_limit,
-             FLAGS_log_shared_gc_flush_replicas_limit,
-             prevent_gc_replicas.size(),
-             fmt::join(prevent_gc_replicas, ", "));
-
-    size_t i = 0;
-    for (const auto &pid : prevent_gc_replicas) {
-        const auto &replica_gc = replica_gc_map.find(pid);
-        if (replica_gc == replica_gc_map.end()) {
-            continue;
-        }
-
-        if (++i > _real_log_shared_gc_flush_replicas_limit) {
-            break;
-        }
-
-        bool mock_flush = false;
-        FAIL_POINT_INJECT_NOT_RETURN_F(
-            "mock_flush_replicas_for_slog_gc", [&mock_flush, this, i](absl::string_view str) {
-                CHECK(buf2bool(str, mock_flush),
-                      "invalid mock_flush_replicas_for_slog_gc toggle, should be true or false: {}",
-                      str);
-                _mock_flush_replicas_for_test = i;
-            });
-        if (dsn_unlikely(mock_flush)) {
-            continue;
-        }
-
-        tasking::enqueue(
-            LPC_PER_REPLICA_CHECKPOINT_TIMER,
-            replica_gc->second.rep->tracker(),
-            std::bind(&replica_stub::trigger_checkpoint, this, replica_gc->second.rep, true),
-            pid.thread_hash(),
-            std::chrono::milliseconds(rand::next_u32(0, FLAGS_gc_interval_ms / 2)));
-    }
-}
-
 void replica_stub::on_gc()
```

Comment:
There is no GC actions in this function, change its name to `collect_statistics` or something else?